### PR TITLE
Fix job cleanup evasion vulnerability, improve job execution error handling

### LINKED ISSUE #612: Errors before cleanup prevent cleanup from running
For example, if `.`, `something/` or `../something` are used as file names in the request, the associated job files will remain stuck. Such names cause errors in the `prime` stage.

### PR DESCRIPTION
**Pull request by Brikaa**
closes #612

### CODE REVIEW

File: api/src/api/v2.js [hunk 1]
```diff
@@ -210,19 +210,26 @@ router.ws('/connect', async (ws, req) => {
...
+                            await job.execute(event_bus);
+                        } catch (error) {
+                            logger.error(
+                                `Error cleaning up job: ${job.uuid}:\n${error}`
+                            );
+                            throw error;
+                        } finally {
+                            await job.cleanup();
+                        }
                         ws.close(4999, 'Job Completed');
```
Comment (Brikaa) [inline api/src/api/v2.js:233]: This line will not execute if an error is thrown above

File: api/src/api/v2.js [hunk 2]
```diff
@@ -276,11 +287,17 @@ router.post('/execute', async (req, res) => {
...
-        await job.cleanup();
-
         return res.status(200).send(result);
     } catch (error) {
-        return res.status(400).json(error);
+        logger.error(`Error executing job: ${job.uuid}:\n${error}`);
+        return res.status(500).send();
+    } finally {
+        try {
+            await job.cleanup();
```
Comment (Brikaa) [inline api/src/api/v2.js:296]: This gets executed before the returns in the try and catch.

File: api/src/api/v2.js [hunk 3]
```diff
@@ -276,11 +287,17 @@ router.post('/execute', async (req, res) => {
...
+        } catch (error) {
+            logger.error(`Error cleaning up job: ${job.uuid}:\n${error}`);
+            return res.status(500).send();
```
Comment (Brikaa) [inline api/src/api/v2.js:299]: On error, this return replaces the return in the outer try/catch

File: packages/bash/5.2.0/build.sh
Comment (Brikaa) [inline packages/bash/5.2.0/build.sh:1]: Unrelated, it wasn't chmod'ed in the repo and got chmod'ed automatically while building.

**Comment by Brikaa** [comment]
GitHub comments I have made on the code are just explanation of the change.

**Review by HexF — APPROVED** [review]

**Pull request merged**